src/apprt/gtk/App.zig

```diff
@@ -1040,6 +1055,44 @@ fn loadRuntimeCss(
     );
 }
 
+fn loadCustomCss(self: *App) Allocator.Error!void {
+    const display = c.gdk_display_get_default();
+
+    // unload the previously loaded style providers
+    for (self.custom_css_providers.items) |provider| {
+        c.gtk_style_context_remove_provider_for_display(
+            display,
+            @ptrCast(provider),
+        );
+        c.g_object_unref(provider);
+    }
+    self.custom_css_providers.clearRetainingCapacity();
+
+    for (self.config.@"gtk-custom-css".value.items) |p| {
+        const path, const optional = switch (p) {
+            .optional => |path| .{ path, true },
+            .required => |path| .{ path, false },
+        };
+        std.fs.accessAbsolute(path, .{}) catch |err| {
+            if (err != error.FileNotFound or !optional) {
+                log.err("error opening gtk-custom-css file {s}: {}", .{ path, err });
+            }
+            continue;
+        };
```

Comment:
I'm not sure if this is the place where this kind of validation should happen. I'm not that familiar with the codebase yet, but maybe this should be validated when the config is loaded, so that the message could be appended to `_diagnostics`?

Reply:
There's a bit of a race condition between the access test and the usage here. I was playing around with this and well and I think that the better option is to open and read the file using Zig functions and then use `gtk_css_provider_load_from_data` or `gtk_css_provider_load_from_bytes`.

We probably should have a helper function that takes a `[]const u8` and then uses `gtk_css_provider_load_from_data` or `gtk_css_provider_load_from_bytes` depending on GTK version (one is deprecated in 4.12 and one is not available until 4.12). That can then be used in all the places that we load CSS.

Reply:
> maybe this should be validated when the config is loaded, so that the message could be appended to `_diagnostics`?

Perhaps when the config is loaded just do an access check and add a diagnostic if the file isn't optional, but I think that's not necessary for this PR.

Reply:
Then I'll leave the scope of this PR as is and won't mess with the validation during config load.